src/Sulu/Bundle/MarkupBundle/Tag/TagRegistry.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\MarkupBundle\Tag;
+
+/**
+ * Container for all tags.
+ */
+class TagRegistry implements TagRegistryInterface
+{
+    /**
+     * @var TagInterface[]
+     */
+    private $tags;
+
+    /**
+     * @param TagInterface[] $tags
+     */
+    public function __construct(array $tags)
+    {
+        $this->tags = $tags;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getTag($name, $type)
+    {
+        if (!array_key_exists($type, $this->tags) || !array_key_exists($name, $this->tags[$type])) {
+            throw new TagNotFoundException($name);
```

Comment:
Sry, have overseen the second condition... But then you should also pass the type here, and adapt the Exception accordingly.